modules/apps/static/portal-osgi-web/portal-osgi-web-portlet-container-test/src/testIntegration/java/com/liferay/portal/osgi/web/portlet/container/upload/test/UploadServletRequestTest.java

```diff
@@ -1294,10 +1294,14 @@ public void shouldNotReturnPreferencesValueWhenModified()
 		public void shouldReturnPreferencesValue() {
 			File tempDir = UploadServletRequestImpl.getTempDir();
 
-			File expectedTempDir = new File(
-				PrefsPropsUtil.getString(
-					PropsKeys.UPLOAD_SERVLET_REQUEST_IMPL_TEMP_DIR,
-					SystemProperties.get(SystemProperties.TMP_DIR)));
+			String tempDirString =
+				UploadServletRequestSettingsUtil.getTempDir();
+
+			if (Validator.isNull(tempDirString)) {
```

Comment:
Just as before, it's quite weird that we are setting the default value here. It was also bad before, although now it's more obvious because of the if clause